### DISCUSSION TITLE
Dev tools on phalcon 5 and PHP 8.1

### DISCUSSION THREAD

**Discussion by jesusfreak3** [phalcon/phalcon#314] [General]
I have recently upgrade to Phalcon V5 and PHP 8.1, and am trying to get devtools up and running.  

If I install from composer dev-tools v5, i see errors related to version (Its now under Support) and the composer file in master of dev tools requires Phalcon V4, not V5.

So I tried switching to dev tools 4.2 which fixed the version error and the dependency of Phalcon V4, however if you attempt to run 'migration run' you start seeing errors.  

They are obvious errors as its looking for Classes that existed in Phalcon 4 and not 5.

ex:  Config is now \Phalcon\Config\Config in V5 but in the Command.php it has the type hint that it should return \Phalcon\Config which fails.

Is there a version of dev tools that actually works with Phalcon V5 /PHP 8?

Thanks

**Comment by jesusfreak3** [comment]
So I decided that probably I would not get an answer very quickly, so I forked dev tools and updated it to support PHP 8 and Phalcon V5 and added a PR for it https://github.com/phalcon/phalcon-devtools/pull/1532  so if anyone is able to review it, maybe this will help others too.  I have updated my projects to use my forked version for now so I can run migrations, models, etc again.

**Reply by borisdelev** [reply]
I need to check but read somewhere for repository with working version. Hmmm will search about it
